estate/models/estate_property.py

```diff
@@ -0,0 +1,113 @@
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('new', 'New'),
+            ('offered received', 'Offer Received'),
+            ('offer accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cnacelled', 'Cancelled')
+            ],
+        default='new',
+        required=True
+    )
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    salesman_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offer")
+    total_area = fields.Float(compute="_compute_total_area")
+    status = fields.Selection(
+        string='Status',
+        selection=[
+            ('new', 'New'),
+            ('offered received', 'Offer Received'),
+            ('offer accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('canceled', 'Canceled')
+            ],
+        default='new',
+        required=True
+    )
+    best_price = fields.Float(compute="_compute_best_offer")
+    date_availability = fields.Date(string="Date Available")
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'A property expected price must be strictly positive'),
+        ('check_selling_price', 'CHECK(selling_price > 0)', 'A property Selling price must be positive'),
+    ]
+
+    @api.constrains('expected_price', 'selling_price')
+    def _not_lower_than_ninty_percent(self):
+        for record in self:
+            if float_is_zero(record.selling_price, precision_rounding=0.01):
+                continue
+            if float_compare(record.selling_price, record.expected_price*0.9, precision_rounding=0.01) == -1:
+                raise ValidationError("The selling price cannot be lower than 90% of the expected price.")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_offer(self):
+        for record in self:
+            record.best_price = max(record.offer_ids.mapped('price'), default=0)
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ""
+
+    def sold_property(self):
+        for record in self:
+            if record.status == 'new':
+                record.status = 'sold'
+            elif record.status == 'canceled':
+                raise UserError("A cancelled property cannot be sold")
+        return True
+
+    def cancel_property(self):
```

Comment:
Action functions must start with `action`
```suggestion
    def action_cancel(self):
```